index.js

```diff
@@ -21,72 +37,98 @@ var resultStrings = [
   'MISMATCH'
 ];
 
-exports.resultCodeToString = function(code) {
-  if (code < 1 || code >= resultStrings.length) { return; }
-  return resultStrings[code];
+HmacAuth.resultCodeToString = function(code) {
+  return resultStrings[code - 1];
 };
 
 function signedHeaders(req, headers) {
-  return headers.map(function(header) { return req.get(header) || ''; });
+  return headers.map(function(header) {
+    var value = req.headers[header];
+    if (typeof value === Array) { value = value.join(','); }
```

Comment:
I think the test passed because `typeof value === Array` fails, so the
`join()` call is skipped, but later the `value` array is being coerced to a
string and `['foo', 'bar'].toString()` => `"foo,bar"` — the same output as
the join method.

On Wed, Oct 7, 2015 at 9:01 PM, Mike Bland <notifications@github.com> wrote:

> In index.js
> <https://github.com/18F/hmac-authentication-npm/pull/1#discussion_r41466852>
> :
>
> >  };
> >
> >  function signedHeaders(req, headers) {
> > -  return headers.map(function(header) { return req.get(header) || ''; });
> > +  return headers.map(function(header) {
> > +    var value = req.headers[header];
> > +    if (typeof value === Array) { value = value.join(','); }
>
> Hmm, it passed my multiply-defined header test, but I agree
> Array.isArray() is the way to roll. Done.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/18F/hmac-authentication-npm/pull/1/files#r41466852>.
>